tv_apps/smart-home/js/home.js

```diff
@@ -187,12 +193,77 @@
           cardManager: that.cardManager
         });
         that._cardPicker.on('hide', that.onCardPickerHide.bind(that));
+        that._cardPicker.on('show', that.onCardPickerShow.bind(that));
+      });
+    },
+
+    initContextMenu: function () {
+      this.moveMenuItem.addEventListener('click', () => {
+          this.edit.toggleEditMode();
+          this.edit.once('arranged', () => this.edit.toggleEditMode());
+          this.edit.toggleArrangeMode();
+      });
+
+      this.renameMenuItem.addEventListener('click', () => {
+        var focus = this.spatialNavigator.getFocusedElement();
+        this.edit.toggleEditMode();
+        this.edit.once('rename-card-ended', () => this.edit.toggleEditMode());
+        this.edit.renameCard(focus, focus.getNodeFromItem(focus.currentItem));
+      });
+
+      this.removeMenuItem.addEventListener('click', () => {
+        var focus = this.spatialNavigator.getFocusedElement();
+        this.edit.toggleEditMode();
+        this.edit.once('delete-card-ended', () => this.edit.toggleEditMode());
+        this.edit.openDeleteCardDialog(focus,
+          focus.getNodeFromItem(focus.currentItem));
+      });
 
-        that.editFolderElem.addEventListener(
-                                        'click', that.onEditFolder.bind(that));
+      this.editFolderMenuItem.addEventListener('click', () => {
+        this._cardPicker.show(this.cardScrollable.currentItem);
       });
     },
 
+    enableContextMenu: function () {
+      var focus = this.spatialNavigator.getFocusedElement();
+      if (this.mode === '' &&
+          (focus === this.cardScrollable || focus === this.folderScrollable)) {
```

Comment:
I think binding context menu on just `folder-list-frame` and `card-list-frame` could be easier, and you don't need to determine everything for switching contextmenu on and off.

But we have some bad layout that prevents us for doing this. So you may need to add some css rule before doing that:
```
#search-bar, #folder-viewer, #card-viewer {
  pointer-events: none;
}

#card-list-frame, #folder-list-frame {
  pointer-events: auto;
}
```